### PR TITLE
feat: add support for multiple rust binaries

### PR DESCRIPTION
**Pull request by pxseu**
This PR adds support for multiple binaries inside a rust project.

- [x] Docs are updated if needed

### CODE REVIEW

**Comment by pxseu** [comment]
In relation to https://github.com/railwayapp/nixpacks/issues/779

File: src/providers/rust.rs [hunk 1]
```diff
@@ -94,22 +94,44 @@ impl RustProvider {
...
                 build.add_cmd(build_cmd);
+<<<<<<< HEAD
                 build.add_cmd(format!("cp target/{target}/release/{workspace} bin"));
             } else {
                 write!(build_cmd, " --target {target}")?;
 
                 if let Some(name) = RustProvider::get_app_name(app)? {
                     build.add_cmd(build_cmd);
                     build.add_cmd(format!("cp target/{target}/release/{name} bin"));
+=======
```
Comment (coffee-cup) [inline src/providers/rust.rs:105]: Please resolve this merge artifact

**Review by coffee-cup — COMMENTED** [review]
Thanks for the PR 😄 
Just one comment regarding a merge conflict but can otherwise merge

File: src/providers/rust.rs (hunk 1 above)
Comment (coffee-cup) [reply src/providers/rust.rs]: Oh nvm it is fixed :)

Comment (pxseu) [reply src/providers/rust.rs]: yep sorry that was an accident on my end cherry picking

**Review by coffee-cup — APPROVED** [review]

**Comment by JakeCooper** [comment]
Thanks @pxseu ❤️🫡

**Pull request merged**